lib_gb/src/machine/registers_handler.rs

```diff
@@ -89,12 +91,12 @@ impl RegisterHandler{
         interrupts_handler.coincidence_interrupt = register & BIT_6_MASK != 0;
 
 
-        if register & 0b11 != ppu.state as u8{
+        if !self.stat_triggered{
```

Comment:
Make sure the stat register could be triggered only once per line

Reply:
The stat interrupt is triggered when it goes from no condition met to any condition met.
So if two interrupts are sets and they right after another, than the first will trigger.
Cause the interrupt wont get to off state but will stay on set state.

Example
Mode 0 amd coincidence are sets and the mode0 stat has triggered and right after it the concidence triggers (which happen in mode 2 right after mode 0) than the concidence wont get triggered.